packages/docusaurus-theme-common/src/utils/codeBlockUtils.ts

```diff
@@ -253,14 +296,35 @@ export function parseLines(
       );
     }
     const metastringRangeClassName = magicComments[0]!.className;
-    const lines = rangeParser(linesRange)
+    rangeParser(linesRange)
       .filter((n) => n > 0)
-      .map((n) => [n - 1, [metastringRangeClassName]] as [number, string[]]);
-    return {lineClassNames: Object.fromEntries(lines), code};
+      .forEach((n) => {
+        lineClassNames[n - 1] = [metastringRangeClassName];
+      });
+    return true;
```

Comment:
I don't see a good reason to move from functional style to imperative side effect style mutating a function input

Please prefer pure functions over functions that mutate their parameters and create side effects. Avoid the word "fill" too, implying that a side effect will happen.